homeassistant/components/light/flux_led.py

```diff
@@ -0,0 +1,116 @@
+"""
+Support for Flux lights.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.flux_led/
+"""
+
+import logging
+import socket
+import voluptuous as vol
+
+from homeassistant.components.light import Light
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['https://github.com/Danielhiversen/flux_led/archive/0.3.zip'
+                '#flux_led==0.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "flux_led"
+
+ATTR_NAME = 'name'
+
+DEVICE_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_NAME): cv.string,
+})
+
+
+def _valid_lights(value):
+    """Validate a dictionary of light definitions."""
+    config = {}
+    for key, device in value.items():
+        config[key] = DEVICE_SCHEMA(device)
+    return config
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required("platform"): DOMAIN,
+    vol.Optional('devices', default={}): vol.All(dict, _valid_lights),
+    vol.Optional('automatic_add', default=False):  cv.boolean,
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """Setup the Flux lights."""
+    import flux_led
+    lights = []
+    light_ips = []
+    for ipaddr, device_config in config["devices"].items():
+        device = {}
+        device["id"] = device_config[ATTR_NAME]
+        device["ipaddr"] = ipaddr
+        light = FluxLight(device)
+        if light.is_valid:
+            lights.append(light)
+            light_ips.append(ipaddr)
+
+    # Find the bulbs on the LAN
+    scanner = flux_led.BulbScanner()
+    scanner.scan(timeout=20)
```

Comment:
Does this mean a 20 second scan on start of HA ? The Hue platform will not scan when devices have been defined. That might be a good idea?